### ISSUE TITLE
The asd-manager can't handle the fact that 2 drives switched driver letters after a reboot

### ISSUE THREAD

**Issue by jtorreke** [openvstorage/alba-asdmanager#299]
This is what the asd-manager logs after 2 devices switched drive letters (sdl is now called sdg):

```
2017-11-08 16:33:08 37100 +0100 - stor-04.ds1 - 28184/140652998293248 - asd-manager/api - 17660 - ERROR - Unexpected exception
Traceback (most recent call last):
  File "/opt/asd-manager/source/app/decorators.py", line 77, in new_function
    return_data = f(*args, **kwargs)
  File "/opt/asd-manager/source/app/api.py", line 94, in list_disks
    DiskController.sync_disks()
  File "/opt/asd-manager/source/controllers/disk.py", line 188, in sync_disks
    DiskController._update_disk(disk, disk_info)
  File "/opt/asd-manager/source/controllers/disk.py", line 204, in _update_disk
    disk.save()
  File "/usr/lib/python2.7/dist-packages/ovs_extensions/dal/base.py", line 172, in save
    prop_values + [self.id])
IntegrityError: UNIQUE constraint failed: disk.name
```

**Comment by jtorreke** [comment]
Temporary workaround:

```
from source.dal.lists.disklist import DiskList
disks = DiskList.get_disks()
for d in disks:
    d.name = '{}_new'.format(d.name)
    d.save()
```
systemctl restart asd-manager

**Comment by JeffreyDevloo** [comment]
Or sync the disks again
from source.controller.disk import DiskController
DiskController.sync_disks()

**Comment by wimpers** [comment]
Can we make the sync_disks() part of the startup sequence of the ASD manager service or would that not help?

**Comment by JeffreyDevloo** [comment]
That is already the case but it won't help with this issue. The asd manager will detect that a drive letter has changed but it will error (hence the workaround from JT)

**Comment by dejonghb** [comment]
FWIW: SAS drives change "drive letters" on almost every reboot, certainly problem in certain setups... (that's why using "drive letters" is a no-no if one needs something that survives reboots and/or adding/removing drives)

**Comment by simonOpenV** [comment]
Fixed in #309 
Package info: openvstorage-sdm_1.11.0-dev.1514981308.eeddedf

**Comment by JeffreyDevloo** [comment]
Part of the https://github.com/openvstorage/alba-asdmanager/releases/tag/1.11.0 release
